### PR TITLE
Adjust initialDashboardId calculation

### PR DESCRIPTION
**Pull request by npfitz**
### Description
Adjusts whether or not we suggest a dashboard as a default location for saving a question. If the most recent thing you have picked in the entity picker is a dashboard, then we will suggest it. Otherwise, we use the existing logic for suggesting a collection

### How to verify
1. Create a question, and save it to a dashboard
2. Create another question and hit save, the dashboard should be suggested as a destination.
3. Instead of saving to the dashboard again, choose a collection and save
4. Now create a 3rd question and hit save, no dashboard should be suggested, and we should fall back to existing `initialCollectionId` logic

### Checklist

- [x] Tests have been added/updated to cover changes in this PR

### CODE REVIEW

File: e2e/test/scenarios/dashboard-filters-2/dashboard-filters-query-stages.cy.spec.ts [hunk 1]
```diff
@@ -1955,7 +1955,11 @@ describe("scenarios > dashboard > filters > query stages + temporal unit paramet
       );
 
       cy.get("@myNewDash").then((dashId: number | any) => {
-        H.visitDashboard(dashId);
+        cy.request("POST", "/api/activity/recents", {
```
Comment (npfitz) [inline e2e/test/scenarios/dashboard-filters-2/dashboard-filters-query-stages.cy.spec.ts:1958]: Should I make an e2e helper for this? I do it a couple times

File: e2e/test/scenarios/native/native.cy.spec.js [hunk 2]
```diff
@@ -58,12 +55,21 @@ describe("scenarios > question > native", () => {
...
         "Third collection",
       );
+      cy.log("after selecting a dashboard, it should be the new suggestion");
+      cy.findByLabelText(/Where do you want to save this/).click();
+    });
 
-      cy.button("Cancel").click();
+    H.entityPickerModal().within(() => {
+      cy.findByText("Orders in a dashboard").click();
+      cy.button("Select this dashboard").click();
```
Comment (npfitz) [inline e2e/test/scenarios/native/native.cy.spec.js:64]: This is a bit of a quirk... you don't actually need to save something to a dashboard to have it be the "last selected thing", you just need to select it via the entity picker.

However, this is a nice secondary assertion that selecting a dashboard from the picker will post to `api/activity/recents`. But I could be convinced that this isn't the best way to do this

File: e2e/test/scenarios/dashboard/dashboard.cy.spec.js [hunk 3]
```diff
@@ -93,6 +93,10 @@ describe("scenarios > dashboard", () => {
       H.queryBuilderHeader().findByText("Save").click();
       cy.findByTestId("save-question-modal").within(modal => {
         cy.findByLabelText("Name").clear().type(newQuestionName);
+        cy.findByLabelText("Where do you want to save this?").should(
```
Comment (npfitz) [inline e2e/test/scenarios/dashboard/dashboard.cy.spec.js:96]: This is a bit of a weird test... It's been updated to say that we should suggest saving the new question in the dashboard that we started from, and we now assert that. But we never actually check that it was made there (though other tests cover this), only that it doesn't show up in the collections view. Maybe that's enough for a smoke test though

File: frontend/src/metabase/dashboard/components/Dashboard/DashboardEmptyState/DashboardEmptyState.tsx [hunk 4]
```diff
@@ -48,6 +48,7 @@ export function DashboardEmptyState({
                     creationType: "custom_question",
                     collectionId: dashboard.collection_id ?? undefined,
                     cardType: "question",
+                    dashboardId: dashboard.id,
```
Comment (npfitz) [inline frontend/src/metabase/dashboard/components/Dashboard/DashboardEmptyState/DashboardEmptyState.tsx:51]: Assuming we want the same logic here as when we create a new question from the "Add Questions" button

**Review by luizarakaki — APPROVED** [review]

File: frontend/src/metabase/components/SaveQuestionForm/context.tsx [hunk 5]
```diff
@@ -82,9 +82,23 @@ export const SaveQuestionProvider = ({
     }
   }, [isLoading]);
 
-  const lastUsedDashboard = recentItems?.find(
-    item => item.model === "dashboard",
-  ) as RecentCollectionItem | undefined;
+  const lastUsedDashboard = useMemo(() => {
```
Comment (sloansparger) [inline frontend/src/metabase/components/SaveQuestionForm/context.tsx:85]: the `lastUsedDashboard` variable name doesn't capture the "ignore the dashboard if there's a more recent collection" part. maybe `defaultDashboard` is a better name?

**Review by sloansparger — APPROVED** [review]

**Pull request merged**